internal/controller/utils.go

```diff
@@ -16,6 +17,35 @@ func updateDeploymentAnnotations(deployment *appsv1.Deployment, annotations map[
 	}
 }
 
+// syncDeploymentReplicas synchronizes the replicas field between two Deployment objects.
+func syncDeploymentReplicas(a, b *appsv1.Deployment) bool {
```

Comment:
Similar to `syncDeploymentResources`, we just need `func setDeploymentReplicas(deployment *appsv1.Deployment, replicas int) (bool, error)`